### DISCUSSION TITLE
Device Service has problems uploading devices to Metadata

### DISCUSSION THREAD

**Discussion by wjoys** [edgexfoundry/community#15] [Q&A] [answered]
Hello everyone,In Jakarta version(C language),I use the example program(examples/counters), add multiple devices in the device file(res/devices/Counters.json), run the device service program, the first run in the function(edgex_metadata_process_device_response) can get the device id.The logs are as follows：

level=INFO ts=2022-06-29T04:00:43Z app=res_temperature msg="Processing Devices from res_temperature/devices"
level=INFO ts=2022-06-29T04:00:43Z app=res_temperature msg="Device res_temperature_710001_AB1100710F000001 created with id e03f1881-b380-43e0-bde3-ca6b6f51fa42"
level=INFO ts=2022-06-29T04:00:43Z app=res_temperature correlation-id=860eaac4-c525-40a4-bb0c-972679efeaba msg="callback: New or updated device res_temperature_710001_AB1100710F000001"
level=INFO ts=2022-06-29T04:00:43Z app=res_temperature msg="Device res_temperature_710002_AB1100710F000002 created with id 99f578d3-49fc-4512-9c80-c06bd6a9b9ff"
.........

After terminating the program and running it again, the function(edgex_metadata_process_device_response) displays an exception log:it does not get the device id, "id(null)" .The log of the exception is as follows:

level=INFO ts=2022-06-29T03:06:47Z app=device-counter msg="Processing Devices from res/devices"
level=INFO ts=2022-06-29T03:06:47Z app=device-counter msg="Device Counter0 created with id (null)"
level=INFO ts=2022-06-29T03:06:47Z app=device-counter msg="Device Counter1 created with id (null)"
......
level=INFO ts=2022-06-29T03:06:47Z app=device-counter msg="Device Counter999 created with id (null)"

The device file(res/devices/Counters.json)  is an array of 1000 devices in the following format:
[
	{
		"name":"Counter0",
		"profileName": "Example-Counter",
		"description": "",
		"protocols":
		{
			"Counter": { "Index": "58" }
		},
		"autoEvents":
		[
			{ "sourceName": "Counter", "onChange": false, "interval": "30s" }
		]
	},
	{
		"name":"Counter1",
		"profileName": "Example-Counter",
		"description": "",
		"protocols":
		{
			"Counter": { "Index": "58" }
		},
		"autoEvents":
		[
			{ "sourceName": "Counter", "onChange": false, "interval": "30s" }
		]
	},
..............
        {
		"name":"Counter999",
		"profileName": "Example-Counter",
		"description": "",
		"protocols":
		{
			"Counter": { "Index": "58" }
		},
		"autoEvents":
		[
			{ "sourceName": "Counter", "onChange": false, "interval": "30s" }
		]
	},
]

Above, why can get the device id  when  run the device service for the first time, but when run it again after terminating the program,it can't get the device id.
       
     Attached are the Device files,Please change the .txt suffix to .json when using
[Counters.txt](https://github.com/edgexfoundry/community/files/9743180/Counters.txt)

     Expect a reply,regards.

**Comment by iain-anderson** [comment, marked as answer]
Hi, I believe this is fixed in https://github.com/edgexfoundry/device-sdk-c/pull/426 so should be working in the next release

**Reply by wjoys** [reply]
Thank you so much !
